java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -840,6 +861,35 @@ private void buildNewClass(NewClassTree tree) {
     build(Lists.reverse(tree.arguments()));
   }
 
+  private void handleExceptionalPaths(Symbol symbol) {
+    TryStatement pop = enclosingTry.pop();
+    TryStatement tryStatement;
+    if(nestedCatchLevel == 0) {
+      tryStatement = pop;
+    } else {
+      tryStatement = enclosingTry.peek();
+    }
+    enclosingTry.push(pop);
+    if(pop != outerTry) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)